src/main/kotlin/com/hj/leetcode/kotlin/problem859/Solution.kt

```diff
@@ -0,0 +1,43 @@
+package com.hj.leetcode.kotlin.problem859
+
+/**
+ * LeetCode page: [859. Buddy Strings](https://leetcode.com/problems/buddy-strings/);
+ */
+class Solution {
+    /* Complexity:
+     * Time O(L) and Space O(1) where L is the length of s;
+     */
+    fun buddyStrings(s: String, goal: String): Boolean {
+        if (s.length != goal.length) {
+            return false
+        }
+
+        val firstThreeDifferences = s.asSequence()
+            .zip(goal.asSequence())
+            .filter { it.first != it.second }
+            .take(3)
+            .toList()
+
+        return when {
+            firstThreeDifferences.isEmpty() -> s.hasRepetition()
+            firstThreeDifferences.size == 2 -> {
+                val diff1 = firstThreeDifferences[0]
+                val diff2 = firstThreeDifferences[1]
+                diff1.first == diff2.second && diff1.second == diff2.first
+            }
+
+            else -> false
+        }
```

Comment:
Make the branch conditions consistent, for example, by testing the size of `firstThreeDifferences` in all cases

Reply:
Addressed at https://github.com/hj-core/hj-leetcode-kotlin/pull/96/commits/67a19d498c6da3389fe75f9df51b3584c8a647e9